src/CoreShop/Component/Address/Context/RequestBased/CountryContext.php

```diff
@@ -46,7 +51,13 @@ public function __construct(RequestResolverInterface $requestResolver, RequestSt
     public function getCountry()
     {
         try {
-            return $this->getCountryForRequest($this->getMasterRequest());
+            $request = $this->getMasterRequest();
+
+            if (null !== $this->country) {
+                return $this->country;
```

Comment:
In currency resolver this is dedicated to a cache resolver. Just my two cents. :) 
However, it's working!

Reply:
ok, it is now separated